### PR TITLE
Safety code testing

### PR DESCRIPTION
**Pull request by pd0wm**

### CODE REVIEW

File: tests/safety/test_toyota.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+        if controls_allowed:
+          send = MIN_ACCEL <= accel <= MAX_ACCEL
+        else:
+          send = accel == 0
+        self.assertEqual(send, self.safety.toyota_tx_hook(self._accel_msg(accel)))
+
+  def test_torque_absolute_limits(self):
+    for controls_allowed in [True, False]:
+      for torque in np.arange(2000, 2000, MAX_RATE_UP):
```
Comment (rbiasini) [inline tests/safety/test_toyota.py]: -2000, 2000, right?

File: tests/safety/test_toyota.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    self.safety.set_torque_meas(500, 500)
+    self.safety.set_desired_torque_last(1000)
+    self.assertFalse(self.safety.toyota_tx_hook(self._torque_msg(1000 - MAX_RATE_DOWN + 1)))
+
+  def test_exceed_torque_sensor(self):
+    self.safety.set_controls_allowed(True)
+
+    for sign in [-1, 1]:
+      self._set_prev_torque(0)
+      for t in np.arange(0, 360, 10):
```
Comment (rbiasini) [inline tests/safety/test_toyota.py]: 360 should be parametrized (MAX_T_ERROR+10)

File: tests/safety/test_toyota.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+MAX_RATE_UP = 10
+MAX_RATE_DOWN = 25
+MAX_TORQUE = 1500
+
+MAX_ACCEL = 1500
+MIN_ACCEL = -3000
+
+MAX_RT_DELTA = 375
+RT_INTERVAL = 250000
```
Comment (rbiasini) [inline tests/safety/test_toyota.py]: Can we import those parameters from the safety_xxx.h file?

File: tests/safety/test_toyota.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    to_push = libpandasafety_py.ffi.new('CAN_FIFOMailBox_TypeDef *')
+    to_push[0].RIR = 0x1D2 << 21
+    to_push[0].RDHR = 0
+
+    self.safety.set_controls_allowed(1)
+    self.safety.toyota_rx_hook(to_push)
+    self.assertFalse(self.safety.get_controls_allowed())
+
+  def test_accel_actuation_limits(self):
+    for accel in np.arange(-4000, 4000, 100):
```
Comment (rbiasini) [inline tests/safety/test_toyota.py]: Generally all those numbers should be parametrized, using the max limit and adding an hardcoded offset.

File: tests/safety/test_toyota.py (hunk 3 above)
Comment (geohot) [reply tests/safety/test_toyota.py]: The test should duplicate them, it's a better test that way

**Pull request merged**